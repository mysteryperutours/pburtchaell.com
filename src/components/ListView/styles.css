.list-view-item {
  margin-top: 1rem;
  margin-bottom: 4rem;
}

/*
 * There should be no padding on the first row of list view items.
 */
.list-view-item-container:nth-of-type(1) .list-view-item,
.list-view-item-container:nth-of-type(1) .list-view-item {
  margin-top: 0;
}
