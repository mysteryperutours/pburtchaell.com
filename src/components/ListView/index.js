import React from 'react';
import PropTypes from 'prop-types';
import Link from 'gatsby-link';
import './styles.css';

/*
 * Function: ListViewItem
 * Description: Renders the title, tags and year of a list view item
 */
const ListViewItemDetails = ({ title, excerpt, category, date }) => (
  <article className="list-view-item">
    <div className="list-view-item-title">
      {title}
    </div>
    <div className="list-view-item-details">
      {excerpt && (
        <div className="list-view-item-excerpt">
          {excerpt}
        </div>
      )}
      <small>{category} &mdash; {date}</small>
    </div>
  </article>
);

ListViewItemDetails.propTypes = {
  title: PropTypes.string.isRequired,
  excerpt: PropTypes.string,
  category: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

ListViewItemDetails.defaultProps = {
  excerpt: null,
};

/*
 * Function: ListViewItem
 * Description: Renders a single linked item in a list of items
 */
const ListViewItem = (props) => {
  const {
    title,
    excerpt,
    category,
    date,
    linkTo,
  } = props;

  return (
    <Link
      to={linkTo}
      title={title}
    >
      <ListViewItemDetails
        title={title}
        excerpt={excerpt}
        category={category}
        date={date}
      />
    </Link>
  );
};

ListViewItem.propTypes = {
  title: PropTypes.string.isRequired,
  excerpt: PropTypes.string,
  category: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  linkTo: PropTypes.string.isRequired,
};

ListViewItem.defaultProps = {
  excerpt: null,
};

export default ListViewItem;
