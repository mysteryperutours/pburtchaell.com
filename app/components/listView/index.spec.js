import React from 'react';
import ListView from './index';

describe(ListView.name, () => {
  it('should render component', () => {
    expect(shallow(
      <ListView
        items={[{
          id: 1
        }, {
          id: 2
        }]}
      />
    )).toMatchSnapshot();
  });
});
