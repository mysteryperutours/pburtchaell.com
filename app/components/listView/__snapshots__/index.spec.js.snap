exports[`ListView should render component 1`] = `
<Row
  className="list-view list-view-type-grid"
  defaultColumn={false}
  elementType="div"
  size="default">
  <ListViewItem
    id={1} />
  <ListViewItem
    id={2} />
</Row>
`;
